util/executil/executil.go

```diff
@@ -21,6 +21,21 @@ const (
 	processGroupTerminationGracePeriod = 3 * time.Second
 )
 
+// HandleExecError wraps an error returned by a function on exec.Cmd, adding
+// stderr to the message if the error is an exec.ExitError.
+func HandleExecError(cmd *exec.Cmd, err error) error {
+	if exitErr, ok := err.(*exec.ExitError); ok {
+		return errors.Wrapf(
+			exitErr,
+			"Failed with exit code %d: %s\n\n%s\n",
+			exitErr.ExitCode(),
+			cmd.String(),
+			string(exitErr.Stderr),
+		)
```

Comment:
The `exitErr` message is usually something like `exit status 1`, in which case the `Failed with exit code 1` is duplicate information.

`string(exitErr.Stderr)` is empty if `cmd.Stderr` was non-nil, resulting in just 3 newlines being appended to the error message.

The following change solves both issues (inspired by https://github.com/cli/cli/blob/0e3c3bb4a428eff7fcff472a34e5208561a9406c/internal/run/run.go#L69-L75) but doesn't print the full command. If you think printing the full command is helpful, feel free to change that.

```suggestion
		stderr := string(exitErr.Stderr)
		if stderr != "" && !strings.HasSuffix(stderr, "\n") {
			stderr += "\n"
		}
		err = fmt.Errorf("%s%s: %w", stderr, cmd.Args[0], err)
		return errors.WithStack(err)
```

Reply:
Thanks!